spec/rubocop/cop/layout/end_of_line_spec.rb

```diff
@@ -1,56 +1,63 @@
 # frozen_string_literal: true
 
 RSpec.describe RuboCop::Cop::Layout::EndOfLine, :config do
+  let(:message)          { 'Carriage return character missing.' }
+  let(:detected_message) { 'Carriage return character detected.' }
+
   shared_examples 'all configurations' do
     it 'accepts an empty file' do
-      inspect_source_file('')
-      expect(cop.offenses.empty?).to be(true)
+      expect_no_offenses('')
     end
   end
 
   shared_examples 'iso-8859-15' do |eol|
     it 'can inspect non-UTF-8 encoded source with proper encoding comment' do
-      inspect_source_file(["# coding: ISO-8859-15#{eol}",
-                           "# Euro symbol: \xa4#{eol}"].join("\n"))
+      inspect_source_file(<<~RUBY)
```

Comment:
Would it be possible to change this to use `expect_offense`?

Reply:
Yeah, it would be.   
I wanted to do it in a separate PR, but we can solve this one right here actually.         
The main goal of this spec - is to test how rubocop works with a file in non-utf8 encoding.  
So reading from a file and respecting magic comment seems like an important part of the process.      

Although maybe it's safe to test it this way: 
```ruby
expect_offense(<<~RUBY.force_encoding('ISO-8859-15'))
   .... 
RUBY
```       
Dealing with the magic comments is not part of rubocop's functionality, so there's no need to test this, we can rely on ruby itself on that. 
What do you think, is the `force_encoding` approach ok?

Reply:
`def expect_offense(source, file = nil, severity: nil, **replacements)`
Oh, I see now.     
Another way would be just to send some temporary file path to `expect_offense` as an argument..

Reply:
Oh, I missed the encoding part. No problem then (maybe add a comment)